packages/app-extensions/src/formField/formField.js

```diff
@@ -30,15 +30,17 @@ export const formFieldFactory = (mapping, data, resources = {}) => {
     const mandatory = !readOnly && _get(modelField, `validation.mandatory`, false)
 
     return (
-      <FormField
-        key={id}
+      <StatedValue
+        dirty={dirty}
+        error={error}
+        hasValue={(value !== null && value !== undefined && value.length !== 0)}
         id={id}
+        immutable={readOnly}
         label={formDefinitionField.label}
         mandatory={mandatory}
         mandatoryTitle={resources.mandatoryTitle}
-        error={error}
         touched={touched}
-        dirty={dirty}
+        type={data.formDefinitionField.componentType}
```

Comment:
i was not aware that you "hardcode" the display componentType into StatedValue. Stated value should not directly be bound to any componentType.